### PR TITLE
Hometasktwo

### PR DESCRIPTION
**Pull request by saitejkuralla**
home task two

### CODE REVIEW

File: HomeTask2/newsFeedScript.js [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+    addMain();
+
+    //Setting data to the elements
+    setData(data);
+}
+
+
+//Header
+function addHeader() {
+
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: remove these spaces

File: HomeTask2/newsFeedScript.js [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+        createHTMLElement(main.htmlElements[i].element, main.htmlElements[i].elementId, main.htmlElements[i].styleClass, main.htmlElements[i].parentElement, main.htmlElements[i].HTMLTextOfElement, main.htmlElements[i].event, main.htmlElements[i].eventFunction);
+    }
+
+    //creation of side panel for adding select category and subscription
+    addSidePanel();
+}
+
+//sidePanel 
+function addSidePanel() {
+
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: remove all unwanted spaces

File: HomeTask2/newsFeedScript.js [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+        createHTMLElement(header.htmlElements[i].element, header.htmlElements[i].elementId, header.htmlElements[i].styleClass, header.htmlElements[i].parentElement, header.htmlElements[i].HTMLTextOfElement, header.htmlElements[i].event, header.htmlElements[i].eventFunction);
+    }
+
+}
+
+//Body
+function addMain() {
+
+    for (let i in main.htmlElements) {
+
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: remove unwanted spaces

File: HomeTask2/newsFeedScript.js [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+            "styleClass": "",
+            "parentElement": "popUpContent",
+            "HTMLTextOfElement": "",
+            "event": "",
+            "eventFunction": ""
+        }
+    ]
+}
+
+
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: remove all these unwanted lines

File: HomeTask2/newsFeedScript.js [hunk 5]
```diff
@@ -0,0 +1,467 @@
+//Create while loading
+window.onload = function () {
+    create();
+};
+//Create Function to create respective element
+function createHTMLElement(element, elementId, styleClass, parentElement, HTMLTextOfElement, event, eventFunction) {
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: avoid passing these many parameters, instead use an object.

File: HomeTask2/newsFeedScript.js [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+    addHeader();
+    //Adding body Part
+    addMain();
+    //Setting data to the elements
+    setData(data);
+}
+//Header
+function addHeader() {
+    for (let i in header.htmlElements) {
+        createHTMLElement(header.htmlElements[i].element, header.htmlElements[i].elementId, header.htmlElements[i].styleClass, header.htmlElements[i].parentElement, header.htmlElements[i].HTMLTextOfElement, header.htmlElements[i].event, header.htmlElements[i].eventFunction);
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: avoid passing individual parameters, instead pass the complete object. same in below element creations aswell

File: HomeTask2/newsFeedScript.js [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+    //Creating and appending p tag to display post description
+    createHTMLElement('p', 'postDescId' + postNumber, 'spacingWithInPosts', 'postContent' + postNumber, dataArray[postNumber - 1].postDesc, '', '');
+    //Creating and appending button tag for displaying popup
+    createHTMLElement('button', 'postContinueBtnId' + postNumber, 'postButton spacingWithInPosts', 'postContent' + postNumber, 'Continue Reading', 'click', continueReading);
+}
+//Setting data and cloning for more posts
+function setData(dataToSet) {
+
+    if (dataToSet.length) {
+        for (let entries = 0; entries < dataToSet.length; entries++) {
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: you can use dataToSet for iterating and avoid using additional variable 'entries'

File: HomeTask2/newsFeedScript.js [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+//sidePanel 
+function addSidePanel() {
+    for (let i in sidePanel.htmlElements) {
+        createHTMLElement(sidePanel.htmlElements[i].element, sidePanel.htmlElements[i].elementId, sidePanel.htmlElements[i].styleClass, sidePanel.htmlElements[i].parentElement, sidePanel.htmlElements[i].HTMLTextOfElement, sidePanel.htmlElements[i].event, sidePanel.htmlElements[i].eventFunction);
+    }
+    for (let i in sharedObj.htmlElements) {
+        createHTMLElement(sharedObj.htmlElements[i].element, sharedObj.htmlElements[i].elementId, sharedObj.htmlElements[i].styleClass, sharedObj.htmlElements[i].parentElement, sharedObj.htmlElements[i].HTMLTextOfElement, sharedObj.htmlElements[i].event, sharedObj.htmlElements[i].eventFunction);
+    }
+    //Creating and appending option tag for displaying categories
+    for (let allCategories = 0; categories[allCategories]; allCategories++) {
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: avoid forloop instead use `categories` itself for iterating.

File: HomeTask2/newsFeedScript.js [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+    }
+    //creation of side panel for adding select category and subscription
+    addSidePanel();
+}
+//sidePanel 
+function addSidePanel() {
+    for (let i in sidePanel.htmlElements) {
+        createHTMLElement(sidePanel.htmlElements[i].element, sidePanel.htmlElements[i].elementId, sidePanel.htmlElements[i].styleClass, sidePanel.htmlElements[i].parentElement, sidePanel.htmlElements[i].HTMLTextOfElement, sidePanel.htmlElements[i].event, sidePanel.htmlElements[i].eventFunction);
+    }
+    for (let i in sharedObj.htmlElements) {
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: please use array helper methods to iterate

File: HomeTask2/newsFeedScript.js [hunk 10]
```diff
@@ -0,0 +1,497 @@
...
+                "parentElement": "postContent" + postNumber,
+                "HTMLTextOfElement": "Continue Reading",
+                "event": "click",
+                "eventFunction": continueReading
+            }
+        ]
+    };
+
+    newsPost.htmlElements.forEach(item => createHTMLElement(item));
+    document.getElementById("postImageId" + postNumber).src = dataArray[postNumber - 1].imageSrc;
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: can we have this in the element creation?

File: HomeTask2/newsFeedScript.js [hunk 11]
```diff
@@ -0,0 +1,497 @@
...
+    let myNode = document.getElementById("content");
+    while (myNode.firstChild) {
+        myNode.removeChild(myNode.firstChild);
+    }
+
+    if (selectedValue != 0) {
+        selectedData = data.filter(function (el) {
+            return el.postCategory == categories[selectedValue];
+        });
+
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: remove this spaces

File: HomeTask2/newsFeedScript.js [hunk 12]
```diff
@@ -0,0 +1,497 @@
...
+    setData(selectedData);
+}
+
+//Email submit event
+function submitEmail() {
+    let arrayOfMailId = [];
+    //getting the mailId entered
+    let mailId = document.getElementById("emailIdTextBox").value;
+    //Regular Expression for email Id
+    let re = /\S+@\S+\.\S+/;
```
Comment (maheshbhuvanagiri) [inline HomeTask2/newsFeedScript.js]: use const where ever its needed.

**Comment by saitejkuralla** [comment]
home task 3 changes merged to master

**Comment by saitejkuralla** [comment]
merge to master branch home task 3

**Pull request merged**